star-wars-films/src/PersonDetail/PersonDetail.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import './PersonDetail.css';
+import PropTypes from 'prop-types';
+
+class PersonDetail extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      person: {}
+    }
+  }
+
+  componentDidMount() {
+    const peopleId = this.props.match.params.peopleId;
+    fetch(`https://swapi.co/api/people/${peopleId}/`)
+      .then(response => response.json())
+      .then(data => {
+        this.setState((prevState, props) => {
+          return {
+            person: data
+          };
+        });
+      })
+      .catch(error => console.log(error));
+  };
+
+  getFilmList(films) {
+    const filmNames = ["A New Hope", "The Empire Strikes Back", "Return of the Jedi", "The Phantom Menace", "Attack of the Clones", "Revenge of the Sith", "The Force Awakens", "The Last Jedi"];
+    let finalFilmList = '';
+    for (let i = 0; i < films.length; i++){
+      const url = films[i];
+      let filmNum = url.split('/').reverse()[1];
+      filmNum = filmNum - 1;
+      finalFilmList += filmNames[filmNum];
+      if (i < (films.length - 1)) {
+        finalFilmList += ", ";
```

Comment:
I had to create a list this more sloppy way as I couldn't get map or forEach to work, see comment in this file.

Reply:
In Home.js for state i would've added films: [].   In Home.js componentDidMount add fetch(https://swapi.co/api/films/) and put it into state films[].  In PersonDetail.js you have person.films array of urls.   You can loop thru that array and search each element in state films[] by url